subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/MutableModuleComponentResolveMetadata.java

```diff
@@ -74,14 +74,15 @@
     ImmutableMap<String, ? extends ConfigurationMetadata> getConfigurations();
 
     /**
-     * Returns the dependency declarations of this component.
+     * Returns a mutable copy of the dependency declarations of this component.
      */
-    List<? extends DependencyMetadata> getDependencies();
+    List<DependencyMetadata> getMutableDependencies();
```

Comment:
❌ We shouldn't have two methods here. This class already has "mutable" in its name, so the dependencies should be mutable by default.

Reply:
That sounds good. Changed it.